src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodFilteringService.cs

```diff
@@ -0,0 +1,191 @@
+                }
+            }
+
+            return results;
+        }
+
+        private static async Task<MultiDictionary<string, string>?> GetPossibleExtensionMethodMatchesInCurrentProcessAsync(Project currentProject, ImmutableHashSet<string> targetTypeNames, bool loadOnly, CancellationToken cancellationToken)
+        {
+            var solution = currentProject.Solution;
+            var graph = currentProject.Solution.GetProjectDependencyGraph();
+            var relevantProjectIds = graph.GetProjectsThatThisProjectTransitivelyDependsOn(currentProject.Id)
+                                        .Concat(currentProject.Id);
+
+            using var syntaxDisposer = ArrayBuilder<SyntaxTreeIndex>.GetInstance(out var syntaxTreeIndexBuilder);
+            using var symbolDisposer = ArrayBuilder<SymbolTreeInfo>.GetInstance(out var symbolTreeInfoBuilder);
+            var peReferences = PooledHashSet<PortableExecutableReference>.GetInstance();
+
+            try
+            {
+                foreach (var projectId in relevantProjectIds.Concat(currentProject.Id))
+                {
+                    // Alway create indices for documents in current project if they don't exist.
+                    loadOnly &= projectId != currentProject.Id;
+
+                    var project = solution.GetProject(projectId);
+                    if (project == null || !project.SupportsCompilation)
+                    {
+                        continue;
+                    }
+
+                    // Transitively get all the PE references
+                    peReferences.AddRange(project.MetadataReferences.OfType<PortableExecutableReference>());
+                    foreach (var document in project.Documents)
+                    {
+                        // Don't look for extension methods in generated code.
+                        if (document.State.Attributes.IsGenerated)
+                        {
+                            continue;
+                        }
+
+                        var info = await document.GetSyntaxTreeIndexAsync(loadOnly, cancellationToken).ConfigureAwait(false);
+
+                        // Don't provide anyting if we don't have all the required SyntaxTreeIndex created.
+                        if (info == null)
+                        {
+                            return null;
+                        }
+
+                        if (info.ContainsExtensionMethod)
+                        {
+                            syntaxTreeIndexBuilder.Add(info);
+                        }
+                    }
+                }
+
+                foreach (var peReference in peReferences)
+                {
+                    var info = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+                        solution, peReference, loadOnly, cancellationToken).ConfigureAwait(false);
+
+                    // Don't provide anyting if we don't have all the required SymbolTreeInfo created.
```

Comment:
same concern here.  i would still provide information.

Reply:
For now, I'm inclined to still do the quick bail, i.e. not return partial results.